akka-http-core/src/main/scala/akka/http/javadsl/settings/ConnectionPoolSettings.scala

```diff
@@ -51,23 +56,32 @@ abstract class ConnectionPoolSettings private[akka] () { self: ConnectionPoolSet
 
   // ---
 
-  def withMaxConnections(n: Int): ConnectionPoolSettings = self.copy(maxConnections = n)
-  def withMinConnections(n: Int): ConnectionPoolSettings = self.copy(minConnections = n)
-  def withMaxRetries(n: Int): ConnectionPoolSettings = self.copy(maxRetries = n)
-  def withMaxOpenRequests(newValue: Int): ConnectionPoolSettings = self.copy(maxOpenRequests = newValue)
+  @ApiMayChange
+  def withHostOverrides(hostOverrides: java.util.List[(String, ConnectionPoolSettings)]): ConnectionPoolSettings = {
+    import scala.collection.JavaConverters._
+    self.copy(hostOverrides = hostOverrides.asScala.toList.map { case (h, s) => ConnectionPoolSettingsImpl.hostRegex(h) -> s.asScala })
+  }
+
+  @ApiMayChange
+  def appendHostOverride(hostPattern: String, settings: ConnectionPoolSettings): ConnectionPoolSettings = self.copy(hostOverrides = hostOverrides :+ (ConnectionPoolSettingsImpl.hostRegex(hostPattern) -> settings.asScala))
+
+  def withMaxConnections(n: Int): ConnectionPoolSettings = self.copyDeep(_.withMaxConnections(n), maxConnections = n)
+  def withMinConnections(n: Int): ConnectionPoolSettings = self.copyDeep(_.withMinConnections(n), minConnections = n)
+  def withMaxRetries(n: Int): ConnectionPoolSettings = self.copyDeep(_.withMaxRetries(n), maxRetries = n)
```

Comment:
Hmm, you'd say so. MiMa doesn't like it, but that looks like it could be a false positive:
```
[error]  * abstract method withMaxRetries(Int)akka.http.javadsl.settings.ConnectionPoolSettings in class akka.http.javadsl.settings.ConnectionPoolSettings does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectAbstractMethodProblem]("akka.http.javadsl.settings.ConnectionPoolSettings.withMaxRetries")
[error]  * in current version there is abstract method withMaxRetries(Int)akka.http.javadsl.settings.ConnectionPoolSettings in class akka.http.javadsl.settings.ConnectionPoolSettings, which does not have a correspondent
[error]    filter with: ProblemFilters.exclude[ReversedAbstractMethodProblem]("akka.http.javadsl.settings.ConnectionPoolSettings.withMaxRetries")
```
Needs a closer look

Reply:
I'd say these can be ignored since this class is not for user extension, so there's no risk of missing implementations.